docs/pbc/all/payment-service-provider/202410.0/base-shop/third-party-integrations/payone/app-composition-platform-integration/payone-acp-app.md

```diff
@@ -87,11 +87,169 @@ When customers pay with PayPal, a shop owner can do the following:
 - Cancel the entire customer order, that is, void the existing preauthorization. In this case, the customer is not charged anything.
 - Cancel one or more items of a customer's order before shipment. The customer is not charged for the canceled items.
 
-## Current limitations
-
-- Payments can be properly canceled only from the the Back Office and not from the Payone PMI.
-- Payments can't be partially canceled. One payment intent is created per order and it can either be authorized or fully cancelled.
-- When an item is canceled on the order details page, all order items are canceled.
+## Implementing PayOne PayPal Express for checkout in a headless application
+
+Use this approach for headless applications with third-party frontends.
+
+### Install modules
+
+Install or upgrade the modules to the specified or later versions:
+- `spryker/kernel-app:1.2.0`
+- `spryker/payment:5.27.0`
+- `spryker/payments-rest-api:1.4.0`
+
+### Pre-order payment flow
+
+When PayOne PayPal Express is integrated into a headless application, orders are processed using a pre-order payment flow:
+
+1. The customer either selects PayPal Express as the payment method at the Product Detail page or the Cart Page.
+2. The `InitializePreOrderPayment` Glue API endpoint (`glue.mysprykershop.com/payments`) is called with the following parameters:
+* Payment provider name: Payone
+* Payment method name: PayPal Express
+* Payment amount
+* Quote data
```

Comment:
TBD: such payment method should not be visible on the Yves Payment checkout page. Currently, there is no different method to get available payment methods for Yves and Glue (it's one PaymentClient/Facade method).

On Yves, it has to be excluded because it is not a regular payment method. But at the same time, it will be excluded from the list of payment methods on Glue: `/checkout-data?include=payment-methods`.

Reply:
I think the front end must be responsible for filtering out what it can't use.